LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/Region.cs

```diff
@@ -234,16 +121,19 @@ public Region(LoRaRegion regionEnum, byte loRaSyncWord, byte[] gFSKSyncWord, (do
         /// Implement correct logic to get the correct transmission frequency based on the region.
         /// </summary>
         /// <param name="upstreamChannel">the channel at which the message was transmitted</param>
-        public double GetDownstreamChannelFrequency(Rxpk upstreamChannel)
+        public bool TryGetDownstreamChannelFrequency(Rxpk upstreamChannel, out double frequency)
         {
+            frequency = 0;
+
             if (this.IsValidRxpk(upstreamChannel))
             {
-                if (this.LoRaRegion == LoRaRegion.EU868)
+                if (this.LoRaRegion == LoRaRegionEnum.EU868)

```

Comment:
Food for thought: shouldn't this kind part of each Region instance, defined during creation? This method will grow for each new region we support.

A possible solution would be to extract this into a class. For instance:

```c#
// create eu region
...
euRegion.DownstreamChannelFrequencyResolver = new SameAsUpstreamChannelFrequencyResolver();

// create us
usRegion.DownstreamChannelFrequencyResolver = new USUpstreamChannelFrequencyResolver();
```